Example of a light shader

ludorverr opened the pull request:
Working with something rather than nothing is always easier when you're new.

clayjohn commented on tutorials/shading/shading_reference/spatial_shader.rst:
```diff
@@ -272,7 +272,17 @@ To write a light function, assign something to DIFFUSE_LIGHT or SPECULAR_LIGHT.
+        DIFFUSE_LIGHT += vec3( dot(NORMAL,LIGHT) * ALBEDO * ATTENUATION );
+    }
+
+Keep in mind DIFFUSE_LIGHT must be added to rather than set, if you don't want one light to overwrite the work of others.
```
"DIFFUSE_LIGHT" should have two back ticks around it ``DIFFUSE_LIGHT``. 

Also, you should rephrase this to be less commanding and more positive. Maybe something like "If you want the lights to add together, add the light contribution to ``DIFFUSE_LIGHT`` using ``+=``, rather than overwriting it."

clayjohn commented on tutorials/shading/shading_reference/spatial_shader.rst:
```diff
@@ -272,7 +272,17 @@ To write a light function, assign something to DIFFUSE_LIGHT or SPECULAR_LIGHT.
-each light type. 
+each light type.
+
+Light shader example:
```
This is fine, but in the rest of the tutorial examples are introduced using a sentence. So, something like "Below is an example of a custom light function using a Lambertian lighting model" might be better.

ludorverr commented:
I've applied your words, feel free to edit anything. If there's a better light shader example you can swap it out too.
I didn't add backticks since previous occurences of DIFFUSE_LIGHT and SPECULAR_LIGHT had no backticks. I've now added backticks for them too.
I also realized the vec3() was pointless so I removed that part.

clayjohn commented:
Looks good!

akien-mga commented:
Thanks!

akien-mga commented:
Cherry-picked for 3.1.
